dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasableResourceInterface.java

```diff
@@ -34,6 +35,9 @@
 import org.apache.http.HttpStatus;
 
 public interface AliasableResourceInterface<T extends Aliasable> {
+    // reserve some prefixes for our own use
+    String[] INVALID_PREFIXES = {"dockstore", "doi", "drs", "trs", "dos", "wes"};
+    String ZENDO_DOI_REGEX = "\\d\\d\\.\\d\\d\\d\\d[\\/-]zenodo\\.\\d*";
```

Comment:
I think you want to have a `^` at the beginning of this regex; otherwise `fooledyou12.1234.zenodo.5678` would be considered invalid when it should be valid. I realize it's extremely unlikely that anybody would try to do this, but still, we should only block actual Zenodo format DOIs.

Also, don't you want the `\\d*` at the end of the expression to be `\\d+`? Or is possible for a Zenodo DOI to not have any digits after the period?

Arguably you should have a `$` a the end of the regex too, but maybe we shouldn't allow `12.1234.zenodo.5678fooledyou`

Reply:
Added ^ to beginning of regex
Added + to end of regex because can be a variable number of digits after the last period
Put pattern outside of method; static not legal for interface
Added $ to end of regex